### PR TITLE
Verlet Cluster Lists SoA Traversal

### PR DESCRIPTION
**Pull request by HobbyProgrammer**
# Description

I added SoA to the verlet cluster lists (to the container, the traversals, and the tests). 

## Related Pull Requests

- #262 Branched from this pull request to have traversals already available.

## Resolved Issues

None

# How Has This Been Tested?

- Unit Tests

### CODE REVIEW

**Comment by HobbyProgrammer** [comment]
The performance is probably not good at the moment, due to the `initClusterTraversal()` and  `endClusterTraversal()` implementations. They load and extract the `SoA`, but since each cluster does not have its own `ParticleCell`, I have to generate a new particle cell for each cluster, then fill it with the particles of the cluster, load or extract SoA, and then copy the particles of the new cell back in its old place.

File: src/autopas/containers/verletClusterLists/VerletClusterLists.h [hunk 1]
```diff
@@ -149,81 +171,206 @@ class VerletClusterLists : public ParticleContainer<Particle, FullParticleCell<P
...
+          loopBody(iClusterStart, _clusterSize, clusterNeighborList);
+        }
+      }
+    }
+  }
+
   /**
    * Recalculate grids and clusters,
    * build verlet lists and
    * pad clusters.
```
Comment (FG-TUM) [inline src/autopas/containers/verletClusterLists/VerletClusterLists.h]: can we merge these doc lines? ;)

File: tests/testAutopas/tests/containers/LinkedCellsVersusVerletClusterListsTest.cpp [hunk 2]
```diff
@@ -5,12 +5,13 @@
...
 #include "LinkedCellsVersusVerletClusterListsTest.h"
+#include "autopas/containers/verletClusterLists/traversals/VerletClustersTraversal.h"
 
-LinkedCellsVersusVerletClusterListsTest::LinkedCellsVersusVerletClusterListsTest()
-    : _verletLists(getBoxMin(), getBoxMax(), getCutoff(), 0.1 * getCutoff(), 2),
-      _linkedCells(getBoxMin(), getBoxMax(), getCutoff(), 0.1 * getCutoff(), 1. /*cell size factor*/) {}
-
+template <autopas::DataLayoutOption dataLayout, bool useNewton3>
 void LinkedCellsVersusVerletClusterListsTest::test(unsigned long numMolecules, double rel_err_tolerance) {
+  Verlet _verletLists{getBoxMin(), getBoxMax(), getCutoff(), 0.1 * getCutoff(), 2};
```
Comment (FG-TUM) [inline tests/testAutopas/tests/containers/LinkedCellsVersusVerletClusterListsTest.cpp]: last one is number of particles per cluster correct? Why 2?

File: tests/testAutopas/tests/selectors/AutoTunerTest.cpp [hunk 3]
```diff
@@ -33,11 +33,26 @@ TEST_F(AutoTunerTest, testAllConfigurations) {
...
+  //                      c01                 with (AoS <=> SoA, noNewton3)             = 2
+  //                      c04                 with (AoS <=> SoA, newton3 <=> noNewton3) = 4
+  //                      c01-combined-SoA    with (SoA, noNewton3)                     = 1
+  // VerletLists:         verlet-lists        with (AoS <=> SoA, newton3 <=> noNewton3) = 4
+  // VerletListsCells:    verlet-sliced       with (AoS, newton3 <=> noNewton3)         = 2
+  //                      verlet-c18          with (AoS, newton3 <=> noNewton3)         = 2
+  //                      verlet-c01          with (AoS, noNewton3)                     = 1
+  // VerletClusterLists:  verlet-clusters     with (AoS <=> SoA, noNewton3)             = 2
+  //                                                                               --------
+  //                                                                                     34
```
Comment (FG-TUM) [inline tests/testAutopas/tests/selectors/AutoTunerTest.cpp]: love it!

File: src/autopas/containers/verletClusterLists/VerletClusterLists.h (hunk 1 above)
Comment (HobbyProgrammer) [reply src/autopas/containers/verletClusterLists/VerletClusterLists.h]: Sure :)

File: tests/testAutopas/tests/containers/LinkedCellsVersusVerletClusterListsTest.cpp (hunk 2 above)
Comment (HobbyProgrammer) [reply tests/testAutopas/tests/containers/LinkedCellsVersusVerletClusterListsTest.cpp]: Correct. I think I just took over the parameters from before. I will remove the "2" to let it use the default of "4".

**Comment by HobbyProgrammer** [comment]
@FG-TUM Ready to merge!

**Pull request merged**